### PR TITLE
Start command

### PR DESCRIPTION
**Pull request by mpetrun5**
Closes: #3 
Closes: #6

### CODE REVIEW

File: cmd/start.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	startCmd.Flags().StringVar(&lbBaseURL, "lb", "", "Target load balancer url (required)")
+	startCmd.Flags().StringVar(&payoutAddress, "payout-address", "", "Payout address to which reward tokens will be sent (required)")
+
+	_ = startCmd.MarkFlagRequired("id")
+	_ = startCmd.MarkFlagRequired("lb")
+	_ = startCmd.MarkFlagRequired("payout-address")
+}
+
+func start(cmd *cobra.Command, _ []string) error {
+	lbURL, err := url.Parse(lbBaseURL)
```
Comment (mpetrunic) [inline cmd/start.go:41]: should you also parse node metrics and rpc url?

File: cmd/start.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		return fmt.Errorf("Failed parsing load balancer url")
+	}
+
+	client := lb.NewClient(lbURL)
+	telemetry := &telemetry.Telemetry{}
+	err = run.Start(client, telemetry, id, nodeRPC, nodeMetrics, payoutAddress)
+
+	if err != nil {
+		return fmt.Errorf("Failed registering to load balancer because of: %v", err)
```
Comment (mpetrunic) [inline cmd/start.go:51]:
```suggestion
		return fmt.Errorf("Failed starting vedran daemon because: %v", err)
```

File: internal/run/run.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"time"
+
+	"github.com/NodeFactoryIo/vedran-daemon/internal/lb"
+	"github.com/NodeFactoryIo/vedran-daemon/internal/telemetry"
+	"github.com/go-co-op/gocron"
+)
+
+// Start registers to load balancer and starts sending telemetry
+func Start(client *lb.Client, telemetry telemetry.TelemetryInterface, id string, nodeRPC string, nodeMetrics string, payoutAddress string) error {
+	err := client.Register(id, nodeRPC, payoutAddress, "test-config-hash")
```
Comment (mpetrunic) [inline internal/run/run.go:14]: you should probably change this so it receives polkadot node client with methods to get hash and metrics.
You can pass same client to startSendingTelemetry method

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: internal/run/run.go (hunk 3 above)
Comment (mpetrun5) [reply internal/run/run.go:14]: Config hash and metrics client are different as they are not on the same url
But I can currently pass metrics client as I have it and will add rpc client when we we have
config hash implementation

File: cmd/start.go (hunk 1 above)
Comment (mpetrun5) [reply cmd/start.go:42]: Added metrics parse
will also add rpc url parse when i create rpc client

File: internal/run/run.go (hunk 3 above)
Comment (mpetrunic) [reply internal/run/run.go]: But that doesn't really matter? You can hide that those are two different urls inside PolkadotNodeService?

**Review by mpetrunic — DISMISSED** [review]

File: internal/telemetry/telemetry.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+const (
+	metricsSendInterval = 30
+	pingSendInterval    = 5
+)
+
+type TelemetryInterface interface {
+	StartSendingTelemetry(scheduler scheduler.Scheduler, lbClient *lb.Client, nodeClient node.Client)
+}
+
+type Telemetry struct{}
```
Comment (MakMuftic) [inline internal/telemetry/telemetry.go:20]: Should be private, only the interface should be exported.

**Review by MakMuftic — CHANGES_REQUESTED** [review]

**Review by MakMuftic — APPROVED** [review]

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**